src/backend/base/langflow/graph/vertex/base.py

```diff
@@ -620,12 +620,29 @@ async def get_result(self, requester: Vertex, target_handle_name: str | None = N
         async with self._lock:
             return await self._get_result(requester, target_handle_name)
 
-    def _log_transaction_async(
+    async def _log_transaction_async(
         self, flow_id: str | UUID, source: Vertex, status, target: Vertex | None = None, error=None
     ) -> None:
-        task = asyncio.create_task(log_transaction(flow_id, source, status, target, error))
-        self.log_transaction_tasks.add(task)
-        task.add_done_callback(self.log_transaction_tasks.discard)
+        """Log a transaction asynchronously with proper task handling and cancellation.
+
+        Args:
+            flow_id: The ID of the flow
+            source: Source vertex
+            status: Transaction status
+            target: Optional target vertex
+            error: Optional error information
+        """
+        async with self._lock:
+            if self.log_transaction_tasks:
```

Comment:
Maybe a few concurrent tasks could be allowed ? By checking the length of log_transaction_tasks.

Reply:
This task is really fast so it should be find for now. 

The number of tasks here is per vertex (per component) so later we can validate how many we can have.